File: catalog/backends/java-quarkus/patterns/InterServiceClient.java
package com.example.service;
import jakarta.enterprise.context.ApplicationScoped;
import org.eclipse.microprofile.config.inject.ConfigProperty;
import java.net.URI;
import java.net.http.*;

// Pattern: Cross-service HTTP call — tracing headers propagated automatically by dd-java-agent.
@ApplicationScoped
public class InterServiceClient {
    private final HttpClient client = HttpClient.newHttpClient();
    @ConfigProperty(name = "TARGET_SERVICE_URL", defaultValue = "http://localhost:8080")
    String targetUrl;
    public String call(String path) throws Exception {
        HttpRequest req = HttpRequest.newBuilder(URI.create(targetUrl + path)).build();
        return client.send(req, HttpResponse.BodyHandlers.ofString()).body();
    }
}

File: catalog/backends/go-gin/patterns/inter_service_client.go
package patterns

import (
	"io"
	"net/http"
	"os"
)

// Pattern: Cross-service HTTP call — tracing headers propagated automatically.
// Orchestrion auto-instruments net/http at compile time.
func CallService(path string) (string, error) {
	baseURL := os.Getenv("TARGET_SERVICE_URL")
	if baseURL == "" {
		baseURL = "http://localhost:8080"
	}
	resp, err := http.Get(baseURL + path)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()
	body, err := io.ReadAll(resp.Body)
	return string(body), err
}
